### PR TITLE
Feature/product

### PR DESCRIPTION
**Pull request by Bisterio**
Review to merge to dev branch

### CODE REVIEW

File: API_v1/Controllers/ProductController.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            return Ok(new BaseResponse { Code = 200, Message = "Get products by seller successfully", Data = response });
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult GetProductById(int id)
+        {
+            Product product = _productService.GetProductById(id);
+            if(product == null)
+            {
+                return Ok(new ErrorResponse { Code = 400, Message = "This product is not exist" });
```
Comment (Baro1509) [inline API_v1/Controllers/ProductController.cs]: This should return NotFound() not Ok()

File: API_v1/Mapper/ProductProfile.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                Origin = product.Origin,
+                PackageMethod = product.PackageMethod,
+                PackageContent = product.PackageContent,
+                Condition = product.Condition,
+                Ratings = product.Ratings,
+                CategoryName = product.Category.Name,
+                MaterialName = product.Material.Name,
+                SellerName = product.Seller.Name
+            };
+        }
```
Comment (Baro1509) [inline API_v1/Mapper/ProductProfile.cs]: What is the purpose of this

File: DataAccess/Implement/ProductDAO.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        public void UpdateProduct(Product product)
+        {
+            Update(product);
+        }
+        public void DeleteProduct(Product product)
+        {
+            Update(product);
+        }
```
Comment (Baro1509) [inline DataAccess/Implement/ProductDAO.cs]: maybe use 1 only? personal reference but this is also ok

**Review by Baro1509 — COMMENTED** [review]
Overall ok, return status is not ok though

File: API_v1/Controllers/ProductController.cs (hunk 1 above)
Comment (Bisterio) [reply API_v1/Controllers/ProductController.cs]: My bad

**Review by Baro1509 — APPROVED** [review]
LGTM

**Pull request merged**